application/src/main/java/org/opentripplanner/model/plan/legreference/ScheduledTransitLegReference.java

```diff
@@ -24,25 +26,28 @@
  * {@link org.opentripplanner.routing.api.RoutingService}
  */
 public record ScheduledTransitLegReference(
-  FeedScopedId tripId,
+  @Nullable FeedScopedId tripId,
   LocalDate serviceDate,
   int fromStopPositionInPattern,
   int toStopPositionInPattern,
   FeedScopedId fromStopId,
 
   FeedScopedId toStopId,
-  FeedScopedId tripOnServiceDateId
+  @Nullable FeedScopedId tripOnServiceDateId
 )
   implements LegReference {
   private static final Logger LOG = LoggerFactory.getLogger(ScheduledTransitLegReference.class);
 
   public ScheduledTransitLegReference {
-    if (tripId != null && tripOnServiceDateId != null) {
+    if ((tripId == null) == (tripOnServiceDateId == null)) {
```

Comment:
Consider: Unsure what to do about this <boolean> == <boolean>. I understand the logic, but it is not common in our code so I had to pause a little. Maybe adding the a static function to `ObjectUtils` is the appropriate action:

```
  public static <T> boolean oneOf(T a, T b) {
    return (a == null) != (b == null);
  }
---
if (!oneOf(tripId, tripOnServiceDateId)) {
```

I do not have a strong opinion - feedback from others is encouraged!